static/js/components/ProfileSelect.tsx

```diff
@@ -0,0 +1,168 @@
+    (profile) => !selected.includes(profile.name)
+  );
+
+  return (
+    <>
+      <Label className="u-sv-1">Apply profiles</Label>
+      <div className="u-text--muted u-sv1">
+        Add profiles and create a hierarchy
+      </div>
+      <Row className="p-profile-select">
+        <Col size={6} className="p-profile-select-box">
+          <i className="p-profile-select-title">Available profiles</i>
+          <hr />
+          <div className="p-profile-select-list">
+            <List
+              items={unselected.map((profile) => (
+                <Fragment key={profile.name}>
+                  <span
+                    className="p-profile-select-name u-truncate"
+                    title={profile.name}
+                  >
+                    {profile.name}
+                  </span>
+                  <div className="u-float-right">
+                    <Button
+                      dense
+                      hasIcon
+                      small
+                      type="button"
+                      onClick={() => toggleProfile(profile)}
+                    >
+                      Apply
+                    </Button>
+                  </div>
+                </Fragment>
+              ))}
+            />
+          </div>
+        </Col>
+        <Col size={6} className="p-profile-select-box">
+          <i className="p-profile-select-title">Applied profiles</i>
+          <hr />
+          <div className="p-profile-select-list">
+            <List
+              items={selected.map((name, i) => {
+                const profile = projectProfiles.find(
+                  (profile) => profile.name === name
+                );
+                if (!profile) {
+                  return null;
+                }
+                return (
+                  <Fragment key={profile.name}>
+                    <span
+                      className="p-profile-select-name u-truncate"
+                      title={profile.name}
+                    >
+                      {profile.name}
+                    </span>
+                    <div className="u-float-right">
+                      <Button
+                        appearance="base"
+                        className="p-profile-select-button"
+                        dense
+                        hasIcon
+                        small
+                        type="button"
+                        disabled={i === 0}
+                        onClick={() => moveUp(profile)}
+                      >
+                        <Icon name="chevron-up" />
+                      </Button>
+                      <Button
+                        appearance="base"
+                        className="p-profile-select-button"
+                        dense
+                        hasIcon
+                        small
+                        type="button"
+                        disabled={i === selected.length - 1}
+                        onClick={() => moveDown(profile)}
```

Comment:
You could pass the `profile` (or just the profile's name) and the index `i` as a parameter to `moveUp` and `moveDown`, and then avoid an unnecessary search in the array inside the function.

Reply:
good idea